src/compiler/types.ts

```diff
@@ -3663,16 +3663,27 @@ namespace ts {
         errors: Diagnostic[];
         wildcardDirectories?: MapLike<WatchDirectoryFlags>;
         compileOnSave?: boolean;
+        configFileSpecs?: ConfigFileSpecs;
     }
 
     export const enum WatchDirectoryFlags {
         None = 0,
         Recursive = 1 << 0,
     }
 
+    export interface ConfigFileSpecs {
+        filesSpecs: string[];
+        includeSpecs: string[];

```

Comment:
Might want to document that `includeSpecs` and `excludeSpecs` are just here for outputting error messages, and everywhere else `validatedIncludeSpecs` and `validatedExcludeSpecs` should be used.